src/main/java/com/bakdata/kafka/KafkaProducerApplication.java

```diff
@@ -110,14 +110,19 @@ protected <K, V> KafkaProducer<K, V> createProducer() {
      */
     protected void runCleanUp() {
         try (final ImprovedAdminClient improvedAdminClient = this.createAdminClient()) {
-            this.cleanUpRun(improvedAdminClient.getSchemaTopicClient());
+            this.cleanUpRun(improvedAdminClient);
         }
     }
 
-    protected void cleanUpRun(final SchemaTopicClient schemaTopicClient) {
+    protected void cleanUpRun(final ImprovedAdminClient improvedAdminClient) {
```

Comment:
Just leave it as it was before and handle the presence of schema registry in SchemaTopicClient